sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -942,21 +945,21 @@ def get_manifest(self, repository: str, tag_or_digest: str, **kwargs) -> GetMani
                 **kwargs
             )
         )
+        if response.http_response.headers['Content-Length'] > MAX_MANIFEST_SIZE:
+            raise ValueError("Manifest size is bigger than 4MB.")
         media_type = response.http_response.headers['Content-Type']
         manifest_bytes = response.http_response.read()
         manifest_json = response.http_response.json()
         if tag_or_digest.startswith("sha256:"):
-            digest = tag_or_digest
-            if not _validate_digest(manifest_bytes, digest):
+            if not _validate_digest(manifest_bytes, tag_or_digest):
                 raise DigestValidationError(
                     "The content of retrieved manifest digest does not match the requested digest."
                 )
-        else:
-            digest = response.http_response.headers['Docker-Content-Digest']
-            if not _validate_digest(manifest_bytes, digest):
-                raise DigestValidationError(
-                    "The server-computed digest does not match the client-computed digest."
-                )
+        digest = response.http_response.headers['Docker-Content-Digest']
```

Comment:
Now you changed the logic: you will check this even tag_or_digest.startswith("sha256:").

Is this by design?

Reply:
Now we changed to validate with the digest in response headers when digest is passing.

Reply:
So we're validating _twice_ as default behaviour? That seem excessive?

Reply:
As discussed offline - validating the data twice is fine, but we should need to compute the digest twice as this is the time consuming part and doesn't need to be duplicated.

Reply:
Yeah, compute digest is wrapped inside _validate_digest(), I'll think about just comparing with two digests.

Reply:
Changed to compare digests here.